src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -329,6 +330,50 @@ void ClusterTaskManager::Heartbeat(bool light_heartbeat_enabled,
   }
 }
 
+bool ClusterTaskManager::PreparePGBundle(BundleSpecification &bundle_spec) {
+  if (pg_bundles_.count(bundle_spec.BundleId()) > 0) {
+    // Duplicate prepare request.
+    return true;
+  }
+
+  std::shared_ptr<TaskResourceInstances> resource_instances =
+      std::make_shared<TaskResourceInstances>();
+  bool allocated = cluster_resource_scheduler_->AllocateLocalTaskResources(
```

Comment:
we use `cluster_resource_scheduler_ ` instead of `scheduling_resource` in new scheduler?

Reply:
Yes, the state of the local node (resource instances) and the node's "view" of the cluster is managed by the cluster resource scheduler.